go/upgrade/api/api.go

```diff
@@ -0,0 +1,92 @@
+// Package api defines the interface exporting the upgrade infrastructure's functionality.
+package api
+
+import (
+	"context"
+	"fmt"
+
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+)
+
+const (
+	UpgradeMethInternal = "internal"
+)
+
+type UpgradeStage uint64
+
+const (
+	UpgradeStageStartup UpgradeStage = iota
+	UpgradeStageConsensus
+
+	upgradeStageAll = (UpgradeStage(1) << iota) - 1
+)
+
+var (
+	DescriptorSchema = []byte(fmt.Sprintf(
```

Comment:
Hm why are we using JSON Schema to validate descriptors here, we are not using it anywhere else? I would suggest the API just takes a `*Descriptor` structure and we can validate it in Go (e.g., add `IsValid`)?